agents/geminidataanalytics/colabs/geminidataanalytics_public_SDK_demo_|_vPublished.ipynb

```diff
@@ -0,0 +1,1538 @@
+        "#               - left_column: str # Join column from left table.\n",
+        "#               - right_column: str # Join column from right table.\n",
+        "# - glossaries: # Business glossary, jargon, etc.\n",
+        "#     - glossary:\n",
+        "#         - term: str # Name of the term. Term can be a word, phrase, abbreviation, etc.\n",
+        "#         - description: str # Description or definition of the term.\n",
+        "#         - synonyms: list[str] # List of synonyms for the term.\n",
+        "# - additional_descriptions:\n",
+        "#     - text: str # Any additional description that was not covered above.\n",
+        "# \"\"\""
+      ],
+      "metadata": {
+        "id": "Bla2oAfRHcOx"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "##################### OPTION 1 ###############################\n",
+        "#######                                                #######\n",
+        "#######        Fill the above YAML template for        #######\n",
+        "#######  system_instruction as shown in example above. #######\n",
+        "#######                                                #######\n",
+        "##############################################################\n",
+        "\n",
+        "system_instruction = \"\"\"\"\"\""
+      ],
+      "metadata": {
+        "id": "S6-3l1ggHK2_"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "##################### OPTION 2 ###############################\n",
+        "#######                                                #######\n",
+        "#######         Leave system_instruction empty         #######\n",
+        "#######         or provide description in form         #######\n",
+        "#######                 of long string.                #######\n",
+        "#######                                                #######\n",
+        "##############################################################\n",
+        "\n",
+        "# Define context for question.\n",
+        "# system_instruction is optional, can be used to steer the agent's answering\n",
+        "# strategy.\n",
+        "system_instruction = \"Think like an Analyst\" # @param {type:\"string\"}"
+      ],
+      "metadata": {
+        "id": "9Ac6KtdMHhea"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### *Helper functions for processing the system_instruction yaml string*"
+      ],
+      "metadata": {
+        "id": "CsiZmq5lOppv"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "import yaml\n",
+        "from yaml.scanner import ScannerError\n",
+        "from collections import Counter\n",
+        "import re\n",
+        "\n",
+        "def cleanup_path(input_string):\n",
```

Comment:
Can you add typing for all of the method parameters and return values? Makes them easier to read.